test/queries_generator/queries_generator.cpp

```diff
@@ -0,0 +1,336 @@
+                aws_expr = "substring(" + aws_expr1 + ", " + aws_expr2 + ", " + aws_expr3 + ")";
+                break;
+	    case 3:
+		aws_expr2 = random_tm_format_string();
+                ceph_expr = "to_string(" + random_query_expr(depth-1, TIMESTAMP, aws_expr1)
+                        + ", \'" + aws_expr2 + "\')";
+                aws_expr = "to_string(" + aws_expr1 + ", \'" + aws_expr2 + "\')";
+                break;
+        }
+    }
+    else if (type == TIMESTAMP)  // return type is TIMESTAMP
+    {
+        string aws_expr1, aws_expr2, date_part;
+        switch (option = rand() % 2)
+        {
+            case 0:
+                date_part = random_date_part();
+                ceph_expr = "date_add(" + date_part + ", " + random_number(aws_expr1) + ", " +
+                        random_query_expr(depth-1, TIMESTAMP, aws_expr2) + ")";
+                aws_expr = "date_add(" + date_part + ", " + aws_expr1 + ", " + aws_expr2 +  ")";
+                break;
+            case 1:
+                ceph_expr = "to_timestamp(\'" + random_timestamp_string(aws_expr1) + "\')";
+                aws_expr = "to_timestamp(\'" + aws_expr1 + "\')";
+                break;
+        }
+    }
+    else if (type == MIX_COL_NUM)
+    {
+        ceph_expr = random_num_col_expr(depth-1, aws_expr);
+    }
+    else if (type == COLUMN)  // return type integer column number
+    {
+        ceph_expr = random_col(aws_expr);
+    }
+    else if (type == NUMBER)  // return type random number
+    {
+        ceph_expr = random_number(aws_expr);
+    }
+    else
+    {
+        aws_expr = "error";
+        ceph_expr = "error";
+    }
+    return ceph_expr;
+}
+
+int main()
+{
+    srand(time(0));
+    int reps, depth;
+    fstream query_file, aws_query_file;
+    query_file.open("ceph_queries.txt", ios::out);
+    aws_query_file.open("aws_queries.txt", ios::out);
+    cout << "Enter number of quries to be generated: ";
+    cin >> reps;
+    cout << "Enter depth of queries to be generated: ";
+    cin >> depth;
+    if(query_file.is_open() && aws_query_file.is_open()) //checking whether the file is open
+    {
+        while (reps)
+        {
+            string aws_expr;
+            int type;
+            string ceph_query = "select ";
+            string aws_query = "select ";
+	    // For multiple projections
+	    //int projection = rand() % 4;
+            //while (projection > 1)
+            //{
+            //    type = rand() % 4;
+            //    ceph_query = ceph_query + random_query_expr(depth,
+            //                    type, aws_expr) + ", ";
+            //    aws_query = aws_query + aws_expr + ", ";
+            //    projection--;
+            //}
+            type = rand() % 4;
+            ceph_query = ceph_query + random_query_expr(depth, type,
+                            aws_expr)+ " from stdin;";
+            aws_query = aws_query + aws_expr + " from s3object;";
```

Comment:
no need for stdin( it's for specific use-cases, not relevant for this comparison)